pandas/core/indexes/base.py

```diff
@@ -4872,7 +4902,23 @@ def _get_engine_target(self) -> ArrayLike:
             # GH#45652 much more performant than ExtensionEngine
             return vals._ndarray
         if type(self) is Index and isinstance(self._values, ExtensionArray):
-            # TODO(ExtensionIndex): remove special-case, just use self._values
+            from pandas.core.arrays import BaseMaskedArray
+
+            if not isinstance(self._values, BaseMaskedArray):
+                # TODO(ExtensionIndex): remove special-case, just use self._values
+                return self._values.astype(object)
+        return vals
+
+    def _get_join_target(self) -> ArrayLike:
+        """
+        Get the ndarray or ExtensionArray that we can pass to the IndexEngine
+        constructor.
```

Comment:
is this docstring still accurate for get_join_target?

Reply:
No, thx